prometheus/rules.yml

```
@@ -158,3 +158,35 @@ groups:
           summary: Watchtower failed when validating CF App configurations
           description: Check the Watchtower logs for failure details.
           icon: ❓
+
+  - name: ElasticsearchAlerts
+    rules:
+      - alert: High disk utilization
+        expr: elasticsearch_filesystem_data_available_bytes / elasticsearch_filesystem_data_size_bytes < 0.2 
```

I think this reads more clearly if we reverse it to be typical fraction `(used space / total space available) > 0.8`. Note - I have not ran/tested this query.
```suggestion
        expr: elasticsearch_filesystem_data_size_bytes / elasticsearch_filesystem_data_available_bytes > 0.8
```

---

that is not how math works. you inverted one size of the equation so you have to invert the other size:
        expr: elasticsearch_filesystem_data_size_bytes / elasticsearch_filesystem_data_available_bytes > 5
I am not sure I like that better.

---

Shouldn't `data_size / space_available` always return a fraction indicating the % of space used? I'm confused now.

---

As written the expression is `free space / total space`

---

Oh! I didn't realize that `elasticsearch_filesystem_data_available_bytes` was the amount of space **left**. I was thinking about it as the amount of space that has been used.

The elasticsearch exporter docs have the following:
```
elasticsearch_filesystem_data_available_bytes	gauge	1	Available space on block device in bytes
elasticsearch_filesystem_data_free_bytes	gauge	1	Free space on block device in bytes
elasticsearch_filesystem_data_size_bytes	gauge	1	Size of block device in bytes
```
The descriptions of these metrics leave much to be desired. How does `data_available_bytes` differ from `data_size_bytes`?

---

Reviewed this, my assumptions were incorrect on the meaning of the metrics. The logic here is sound.